python/fledge/common/configuration_manager.py

```diff
@@ -294,12 +294,16 @@ def get_entry_val(k):
                     d = {entry_name: entry_val}
                     expected_item_entries.update(d)
                 num_entries = expected_item_entries.get(entry_name)
+                if num_entries is None:
+                    _logger.warning('For {} category, DISCARDING unrecognized entry name {} for item name {}'
+                                    .format(category_name, entry_name, item_name))
+                    try:
+                        del category_val_copy[item_name][entry_name]
+                    except RuntimeError:
```

Comment:
It should catch general exception here instead of RunTimeError. Even in this case I believe only it throws KeyError